internal/pkg/deploy/pipeline.go

```diff
@@ -637,6 +751,33 @@ func (stg *PipelineStage) buildDeploymentsGraph() *graph.Graph[string] {
 	return digraph
 }
 
+func (stg *PipelineStage) buildPrePostDeploymentsGraph(depType string) *graph.Graph[string] {
```

Comment:
I think we can reduce the duplicacy in `buildDeploymentsGraph()` and `buildPrePostDeploymentsGraph(depType string)` by having something like this. 
```
func (stg *PipelineStage) buildDeploymentsGraph() *graph.Graph[string] {
	return stg.buildGraphFromDeployments(stg.deployments)
}

func (stg *PipelineStage) buildPrePostDeploymentsGraph(depType string) *graph.Graph[string] {
	var deployments manifest.PrePostDeployments
	switch depType {
	case "pre":
		deployments = stg.preDeployments
	case "post":
		deployments = stg.postDeployments
	}
	return stg.buildGraphFromDeployments(deployments)
}

func (stg *PipelineStage) buildGraphFromDeployments(deployments manifest.PrePostDeployments) *graph.Graph[string] {
	var names []string
	for name := range deployments {
		names = append(names, name)
	}
	digraph := graph.New(names...)
	for name, conf := range deployments {
		if conf == nil {
			continue
		}
		for _, dependency := range conf.DependsOn {
			digraph.Add(graph.Edge[string]{
				From: dependency, // Dependency must be completed before name.
				To:   name,
			})
		}
	}
	return digraph
}
```